crates/turborepo-lib/src/run/mod.rs

```diff
@@ -61,7 +62,7 @@ impl Run {
         }
 
         // There's some warning handling code in Go that I'm ignoring
-        if self.base.ui.is_ci() && !opts.run_opts.no_daemon {
+        if UI::is_ci() && !opts.run_opts.no_daemon {
```

Comment:
Remove the `UI::is_ci` variant to discourage others from depending on it. If we think this behavior should be used more I'd suggest we add a new `is_ci_and_no_tty`
```suggestion
        if turborepo_ci::is_ci() && !std::io::stdout().is_terminal() && !opts.run_opts.no_daemon {
```